File: TwoSum/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
       
        indices = {}

        for i in range(len(nums)):
            indices[nums[i]] = i

        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in indices and indices[diff] != i:
                return [i,indices[diff]]

File: Two Sum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> umap;
        for (int i = 0; i < nums.size(); i++){
            umap[nums[i]] = i;
        }

        for (int i = 0 ; nums.size(); i++){
            int diff = target - nums[i];
            if (umap.count(diff) && umap[diff] != i){
                return {i, umap[diff]};
            }
        }
    
        return {0, 0};
    }
};

